android/src/XCSoar.java

```diff
@@ -103,14 +103,15 @@
     getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN|
                          WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
 
-    /* Workaround for layout problems in Android KitKat with immersive full
-       screen mode: Sometimes the content view was not initialized with the
-       correct size, which caused graphics artifacts. */
-    getWindow().addFlags(WindowManager.LayoutParams.FLAG_LAYOUT_IN_SCREEN|
-                         WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS|
-                         WindowManager.LayoutParams.FLAG_LAYOUT_INSET_DECOR|
-                         WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
-
+    if (!Loader.isLegacy) {
```

Comment:
Explanation missing in your commit message

Reply:
Will add source code comment.

Reply:
But this is the wrong check. `isLegacy` checks whether the "legacy" library was loaded; it does not check the Android version. Of course, the Android version determiners whether the "legacy" library should be loaded, but I don't like this indirection; it is fragile and non-obvious.
If you really believe this code should be disabled, check `android.os.Build.VERSION.SDK_INT`.
Look at https://github.com/XCSoar/XCSoar/commit/b1f8d0ccf1dc6cdc3d2c5f21191667237237f556#diff-58a404defd924a09562ddea6a3f1cc86L112